src/docs/arch-disagg-layered.dox

```diff
@@ -0,0 +1,107 @@
+
+These semantics are achieved by using two underlying constituents: a shared stable table and an
+ingest table. The node responsible for writing to shared storage operates directly on a single
+shared table and has the guarantee that no other node will perform writes to that table.
+All other nodes will also have an ingest table, which holds a short-term view of recent changes.
+
+This view is layered over the shared stable table, so any operations on a node with both an ingest
+and stable table will first look for a result in the ingest table, and if no result is found, fall
+back on the shared table.
+
+Layered tables are primarily for Disaggregated Storage Clusters (DSC) but can also be used outside
+of DSC for testing purposes, in particular, via local Page and Log Mock (PALM) infrastructure.
+
+\image html shared_cluster.png
+
+Stable Table:
+- A table shared across nodes in the cluster
+- Backed by a block manager that uses shared storage and supports having multiple different
+WiredTiger nodes share the same underlying data structure
+
+Ingest Table:
+- A short-term in-memory table that stores data that has been changed more recently than is
+available in the stable table
+- Supersedes checkpointed content in the stable table
+- This table is specific to a particular node
+
+@section layered-api-restrictions API restrictions
+
+Layered tables support all the regular tables operations, with exceptions listed here:
+
+- All writes, from all replicas, must be timestamped
+- All nodes must use the same URI for the same table
+- All nodes must use byte-wise identical keys, values, and timestamps for the same records.
+- Prepared transactions will behave as described in the design for distributed transactions in
+disaggregated storage
+- Durability is provided by the logical log service, not WiredTiger
+    - Commit durability (using write-ahead log) is not supported for Layered Tables.
+    - Checkpoint durability is asynchronous. The completion of a `WT_SESSION::checkpoint()` call
+    does not guarantee that the resulting checkpoint is on storage and visible to other nodes yet.
+- Table compaction is not supported on layered tables
+- Backup cursors will not return information about layered tables
+- Bulk cursors are not supported on layered tables
+
+@section shared-tables Shared tables
+
+A shared table is a table whose data is stored in a shared storage back-end. A table
+that stores all its data on disk can be called a local or regular table. There is no requirement for
+a stable table to be a shared table, but in all known cases, it is. The ingest table is a local table.
+
+There are also shared tables that are not part of any layered table (e.g., a shared history store
+table or a shared metadata table). The key difference between a regular table and a shared table is
+that the shared one uses the disaggregated block manager.
+
+Generally speaking, a layered table is not a table in the traditional sense, but rather an adapter
```

Comment:
You could move all of this to immediately after you introduce the two terms in the sentence `These semantics are achieved by using two underlying constituents: a shared stable table and an ingest table.`

Reply:
Is that related to this? https://github.com/wiredtiger/wiredtiger/pull/12350#discussion_r2328959183

If so, then I moved it. Please check